src/pages/edit-profile/EditProfile.tsx

```diff
@@ -103,6 +104,63 @@ const EditProfile = () => {
     }
   }, [loading, profileState, initialEditProfileState])
 
+  const changedProfileFields = useMemo<Partial<EditProfileState>>(() => {
+    if (!initialEditProfileState || !profileState) return {}
+
+    const changedInitial = {
+      firstName: initialEditProfileState.firstName,
+      lastName: initialEditProfileState.lastName,
+      photo: initialEditProfileState.photo,
+      city: initialEditProfileState.city,
+      country: initialEditProfileState.country,
+      nativeLanguage: initialEditProfileState.nativeLanguage,
+      professionalSummary: initialEditProfileState.professionalSummary,
+      videoLink: initialEditProfileState.videoLink
+    }
+
+    const changedCurrent = {
+      firstName: profileState.firstName,
+      lastName: profileState.lastName,
+      photo: profileState.photo,
+      city: profileState.city,
+      country: profileState.country,
+      nativeLanguage: profileState.nativeLanguage,
+      professionalSummary: profileState.professionalSummary,
+      videoLink: profileState.videoLink
+    }
+
+    const hasPhotoChanged = hasPhotoChanges(
+      changedInitial.photo,
+      changedCurrent.photo
+    )
+
+    const hasChanged =
+      hasChanges(changedInitial, changedCurrent) || hasPhotoChanged
+
+    if (hasChanged) {
+      const changes: Partial<EditProfileState> = {
+        ...changedCurrent
+      }
+
+      if (!hasChanges(changedInitial.videoLink, changedCurrent.videoLink)) {
+        delete changes.videoLink
+      }
+
+      if (hasPhotoChanged) {
+        changes.photo = changedCurrent.photo
+      }
+
+      return changes
+    } else {
+      return {}
+    }
```

Comment:
Try to replace this block with `if-not` approach. You will not have an exta `else` here.